examples/fakeserver/src/main/java/com/stellarstation/api/fakeserver/FakeStellarStationService.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 Infostellar, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stellarstation.api.fakeserver;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.Timestamps;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.stellarstation.api.v1.ReceiveTelemetryResponse;
+import com.stellarstation.api.v1.SatelliteStreamRequest;
+import com.stellarstation.api.v1.SatelliteStreamResponse;
+import com.stellarstation.api.v1.StellarStationServiceGrpc.StellarStationServiceImplBase;
+import com.stellarstation.api.v1.Telemetry;
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import java.time.Clock;
+import java.time.Duration;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+
+/**
+ * A fake implementation of {@link StellarStationServiceImplBase} which:
+ *
+ * <ul>
+ *   <li>Returns 1MB of random telemetry payload every second.
+ *   <li>Echos received comand payloads back as telemetry payloads.
+ *   <li>Cancels the stream after 5 minutes.
+ * </ul>
+ *
+ * This functionality should allow API clients to check send/receive functionality along with
+ * recovering from errors (e.g., reconnecting the stream).
+ */
+class FakeStellarStationService extends StellarStationServiceImplBase {
+
+  @Inject
+  FakeStellarStationService() {}
+
+  @Override
+  public StreamObserver<SatelliteStreamRequest> openSatelliteStream(
+      StreamObserver<SatelliteStreamResponse> responseObserver) {
+    ServiceRequestContext ctx = RequestContext.current();
+    ctx.setRequestTimeout(Duration.ZERO);
+    ctx.setMaxRequestLength(0);
+    var future =
```

Comment:
Can you use an explicit type unless that's too verbose?

Reply:
I wasn't sure what the type is other than "something with a cancel method", so thought it's a good use case of `var`. But if it's important for readers than makes sense to be explicit (I guess very few people know what a `ScheduledFuture` is though).